keps/sig-network/3705-cloud-node-ips/README.md

```diff
@@ -208,25 +215,48 @@ will try to parse it as a single IP address, fail, log an error, and
 leave the node in the tainted state, which is exactly what we wanted
 it to do if it can't interpret the `--node-ip` value correctly.
 
-Therefore, we do not need a new annotation for the new `--node-ip`
+Therefore, we do not _need_ a new annotation for the new `--node-ip`
 values; we can continue to use the existing annotation, assuming
 existing cloud providers will treat unrecognized values as errors.
 
-```
-<<[UNRESOLVED annotation-name ]>>
+That said, the existing annotation name is
+`alpha.kubernetes.io/provided-node-ip` but it hasn't been "alpha" for
+a long time. We should fix this. So:
+
+  1. When the `CloudNodeIPs` feature is enabled, kubelet will set both
+     the legacy `alpha.kubernetes.io/provided-node-ip` annotation and
+     the new `kubernetes.io/provided-node-ip` annotation. (It always
+     sets them both to the same value, even if that's a value that old
+     cloud providers won't understand).
+
+  2. When the `CloudNodeIPs` feature is enabled, the cloud provider
+     will use the new `kubernetes.io/provided-node-ip` annotation _if
+     the legacy alpha annotation is not set_. (But if both annotations
+     are set, it will prefer the legacy annotation, so as to handle
+     rollbacks correctly.)
+
+  3. A few releases after GA, kubelet will stop setting, and begin
+     deleting, the legacy annotation, and switch to only using the new
+     annotation. Cloud providers will also switch to only using the
+     new annotation, and perhaps logging a warning if they see a node
+     with the old annotation but not the new annotation.
+
+<<[UNRESOLVED new-annotation-name ]>>
 
-The annotation name is `alpha.kubernetes.io/provided-node-ip`
-but it hasn't been "alpha" for a long time. Should we rename it? In
-that case, we probably need to keep supporting both versions for a
-while.
+Tim suggested "we should probably make clear this is the
+kubelet-provided-node-ip ?". But it's really more the
+administrator-provided node IP? (The administrator provides it to
+kubelet, who provides it to the cloud-provider.) But just
+"provided-node-ip" seems fine to me either way?
```

Comment:
I guess I was hoping for some clarity as to where/who provided it.

kubelet-provided-node-ip or node-ip-override or something seem clearer to me, but I don't feel very strongly, so if you disagree, just nix this.

Take a look at https://kubernetes.io/docs/reference/labels-annotations-taints/ and see if it could be improved?

Reply:
To me, calling it `kubelet-provided-node-ip` implies "kubelet provided this node IP", but that feels wrong; yeah, kubelet created the annotation, but it didn't come up with the _value_ itself. It's just the messenger.

I'm ok with `node-ip-override`. If you want to emphasize who created the annotation then maybe `kubelet.kubernetes.io/whatever-node-ip-whatever`?

Reply:
> implies "kubelet provided this node IP"

yeah, that's not correct.

I can live with it, since I now know what it is and it *is* documented.  If you are now annoyed by it, I'd be good with `node-ip-override` too but I am not going to argue for it any further :)